### ISSUE TITLE
Add `aws_aurora`

### ISSUE THREAD

**Issue by depombo** [iasql/iasql-engine#1121]

**Comment by dfellis** [comment]
Digging into this task:

1. Aurora is treated by the SDK as just another feature of RDS (though with wildly non-overlapping configuration knobs). Whether we should break this out into a separate `aws_aurora` module or make it part of `aws_rds` is something we need to think about.
2. There's an Aurora v1 and v2, with [v1 only available in certain regions with fewer features](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-serverless.html#aurora-serverless.limitations) and having [some special flags just for itself](https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/clients/client-rds/interfaces/createdbclustercommandinput.html#enablehttpendpoint). It also seems [Aurora v1 MySQL is deprecated at the end of February next year](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Aurora.MySQL56.EOL.html) while [Aurora v1 Postgres was deprecated at the end of January this year while another batch of deprecation (of old Postgres versions on Aurora v2) is happening February next year](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraPostgreSQLReleaseNotes/AuroraPostgreSQL.Updates.html).

I think we should not support Aurora v1 at all, but I am still not sure if it should be a separate module or not.

Relatedly, some of the complexity here is that Aurora is intertwined with multi-region RDS, so like `aws_dynamo` it seems unlikely that we can fully implement this module until we refactor everything for multi-region support.

**Comment by aguillenv** [comment]
Using the AWS console, aurora instances are listed as RDS instances or are they different resources? I think the answer to that question is the answer if they should be different modules or not. We have other modules that use the same SDK API behind scenes like `aws_vpc`, `aws_security_groups` or `aws_ec2` all using AWS EC2 API.

I agree we should only support v2.

**Comment by dfellis** [comment]
So, Amazon has Aurora inside of RDS in the dashboard as a special database type with two configurations. It also automatically requires you to name the "DB cluster" instead of the DB instance like the others, but that's subtle, so that implies this task should go into the `aws_rds` module.

![Image](https://user-images.githubusercontent.com/765551/185438272-5fd639ce-27ff-4455-b52b-fc8339f343ee.png)

The only thing that's got me still considering an alternative is the significant *lack* of overlap between Aurora configuration properties and "normal database" configuration properties. Their API is ungainly the way it is structured and having a ton of `NULL` configuration columns for users (whether they're using Aurora or not) feels bad to me.

Possibly it could be a special `aurora` table inside of the `aws_rds` module, but then why require "regular" RDS users to deal with it?

**Comment by dfellis** [comment]
Why `Aurora` is not given top-level billing while `Dynamo` is makes no sense to me. AWS's marketing is totally inconsistent, just like their APIs, but we wanted to keep our groupings similar to how AWS themselves do it so we have less to explain to users, so putting in new `aurora` tables (I'm thinking `aurora_cluster` and `aurora_instance` with the latter being not required if they chose serverless aurora) is how it should be structured. I guess?

![I GUESS](https://pbs.twimg.com/media/Ce8dyvjW8AIP71I.jpg)

**Comment by depombo** [comment]
Given all the information you provided here, I agree with your conclusion to make aurora tables within rds

**Comment by aguillenv** [comment]
Taking a look at the console, in the end, aurora is not just any other rds db engine? I'm doubting why we need a module or special table for it. I understand that it requires a cluster, and taking a look I see that MySQL too. Why not add a `cluster` FK to the current rds instance table and make a constraint that the cluster cannot be null for aurora or MySQL? Or does aurora have special API calls?

**Comment by yrobla** [comment]
I agree, i see conceptually as another RDS type, maybe with more specific configs... but at the end i think it should be under RDS

**Comment by aguillenv** [comment]
ohh I see, then if cluster command and instance commands are different for aurora in general, I'd lean towards the `aws_rds_aurora` module. Also, we should add a filter in current `aws_rds` to not load aurora instances if it is doing that now.
